test/extensions/emotable.ts

```diff
@@ -91,3 +114,80 @@ describe('RMRKMultiAssetEmotableMock', async function () {
     });
   });
 });
+
+describe('RMRKEmoteTrackerMock', async function () {
+  let emoteTracker: RMRKEmoteTrackerMock;
+  let tokenA: ERC721Mock;
+  let tokenB: ERC721Mock;
+  let owner: SignerWithAddress;
+  let addrs: SignerWithAddress[];
+  const tokenId = bn(1);
+  const emoji1 = Buffer.from('😎');
+  const emoji2 = Buffer.from('😁');
+
+  beforeEach(async function () {
+    [owner, ...addrs] = await ethers.getSigners();
+    ({ emoteTracker, tokenA, tokenB } = await loadFixture(emoteTrackerFixture));
+  });
+
+  it('can support IERC165', async function () {
+    expect(await emoteTracker.supportsInterface(IERC165)).to.equal(true);
+  });
+  it('can support IRMRKEmoteTracker', async function () {
+    expect(await emoteTracker.supportsInterface(IRMRKEmoteTracker)).to.equal(true);
+  });
+
+  it('does not support other interfaces', async function () {
+    expect(await emoteTracker.supportsInterface(IOtherInterface)).to.equal(false);
+  });
+
+  describe('With minted tokens', async function () {
+    beforeEach(async function () {
+      await tokenA.mint(owner.address, 1);
+      await tokenA.mint(owner.address, 2);
+      await tokenB.mint(owner.address, 1);
+      await tokenB.mint(owner.address, 2);
+    });
+
+    it('can emote', async function () {
+      await expect(emoteTracker.emote(tokenA.address, tokenId, emoji1, true))
+        .to.emit(emoteTracker, 'Emoted')
+        .withArgs(owner.address, tokenA.address, tokenId.toNumber(), emoji1, true);
+      expect(await emoteTracker.getEmoteCount(tokenA.address, tokenId, emoji1)).to.equal(bn(1));
+    });
+
+    it('can undo emote', async function () {
+      await emoteTracker.emote(tokenA.address, tokenId, emoji1, true);
+
+      await expect(emoteTracker.emote(tokenA.address, tokenId, emoji1, false))
+        .to.emit(emoteTracker, 'Emoted')
+        .withArgs(owner.address, tokenA.address, tokenId.toNumber(), emoji1, false);
+      expect(await emoteTracker.getEmoteCount(tokenA.address, tokenId, emoji1)).to.equal(bn(0));
+    });
+
+    it('can be emoted from different accounts', async function () {
```

Comment:
Should we add a case verifying that the same address can emote different emojis on a same token?

Reply:
Sounds good, added it on the last commit.